contracts/PoolManager.sol

```diff
@@ -338,26 +337,26 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC1155, IERC1155Rec
         _accountDelta(currency, -(paid.toInt128()));
     }
 
-    function _burnAndAccount(Currency currency, uint256 amount) internal {
-        _burn(address(this), currency.toId(), amount);
-        _accountDelta(currency, -(amount.toInt128()));
-    }
-
-    function onERC1155Received(address, address, uint256 id, uint256 value, bytes calldata) external returns (bytes4) {
-        if (msg.sender != address(this)) revert NotPoolManagerToken();
-        _burnAndAccount(CurrencyLibrary.fromId(id), value);
-        return IERC1155Receiver.onERC1155Received.selector;
+    /// @inheritdoc IPoolManager
+    function burn(Currency currency, uint256 amount) external noDelegateCall onlyByLocker {
+        _burnAndAccount(currency, amount);
     }
 
-    function onERC1155BatchReceived(address, address, uint256[] calldata ids, uint256[] calldata values, bytes calldata)
+    /// @inheritdoc IPoolManager
+    function batchBurn(Currency[] calldata currencies, uint256[] calldata amounts)
```

Comment:
since we have batchBurn i wonder if we could/should also have batchMint? or else have neither.. extra warm calls back into PM are quite cheap so its not horrible to do batch burn externally

Reply:
I added batchBurn just to keep parity with before which has the batch1155OnReceived but I agree maybe it's not necessary since like you said warm calls are cheappp

Reply:
FWIW I don't think batchMint is useful so yea maybe lets take out

Reply:
I think batch burn only existed bc 1155 has batch transfer so we had to implement onbatchreceived

Reply:
Removed along with burnAndAccount